include/albatross/src/core/distribution.hpp

```diff
@@ -99,8 +99,17 @@ struct MarginalDistribution : public DistributionBase<MarginalDistribution> {
     return covariance.diagonal()[i];
   }
 
+  bool approximately_equal(const MarginalDistribution &other,
+                           const double epsilon = 1e-3) const {
```

Comment:
nit: let's pull out the default value:
```
constexpr double default_approximately_equal_epsilon = 1e-3;
```
and remove the `const` since we're passing in `epsilon` by value.
```suggestion
                           double epsilon = default_approximately_equal_epsilon) const {
```